src/sniffer/InterfaceHandler.c

```diff
@@ -0,0 +1,114 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <stdbool.h>
+#include <pcap.h>
+
+#include "InterfaceHandler.h"
+#include "ErrorHandler.h"
+#include "Sniffer.h"
+
+pcap_if_t *InterfaceHandler_availableInterfaces;
+pcap_if_t *InterfaceHandler_currentInterface;
+pcap_t *InterfaceHandler_captureHandler;
+
+void InterfaceHandler_Init() {
+    // Find list of interfaces available in the system
+    if(pcap_findalldevs(&InterfaceHandler_availableInterfaces, ErrorHandler_GetPcapErrorBuffer()) == PCAP_ERROR) {
+        ErrorHandler_DisplayPcapErrorAndExit("Error scanning for devices", true);
+    }
+}
+
+InterfaceHandler_status_t InterfaceHandler_SelectInterface(int interfaceNumber) {
+    int index;
+    pcap_if_t *currentIf;
+
+    if (InterfaceHandler_availableInterfaces == NULL) {
+        ErrorHandler_DisplayErrorAndExit("[InterfaceHandler] Interfaces not available!");
+    } else {
+        for (currentIf = InterfaceHandler_availableInterfaces, index = 0; currentIf != NULL; currentIf = currentIf->next, index++) {
```

Comment:
Since there are multiple places in which we need to run through the linked list (LL), what about creating a wrapper function to which we pass a function pointer such that, inside the wrapper we run through the LL and the function is called on every node of the LL?

Reply:
We can do it, but I think that for the two 2-3 lines for loops we currently have it is a little bit overkill. It would add unnecessary complexity to two very simple loops don't you think?
Maybe if we had more complex loop bodies and/or had it in more places it would be beneficial.

Reply:
Agreed, let's just leave it for the future in case we need it.